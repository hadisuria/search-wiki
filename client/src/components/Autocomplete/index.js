import React from "react";
import ReactAutocomplete from "react-autocomplete";
import Input from "../Input";

const Autocomplete = ({ articles, searchValue, onSearchChange }) => {
	console.log("search value", searchValue);
	return (
		<ReactAutocomplete
			items={articles}
			renderInput={Input}
			inputProps={{ placeholder: "Input text here..." }}
			getItemValue={(item) => item.label}
			renderMenu={(children, value, style) => {
				return articles && articles.length ? (
					<div style={{ ...style }} className="input-suggestion">
						{children}
						<a href={`/search?query=${value}`} className="search-link">
							See all result
						</a>
					</div>
				) : (
					<></>
				);
			}}
			renderItem={(item, highlighted) => (
				<div
					key={item.id}
					style={{ backgroundColor: highlighted ? "#eee" : "transparent" }}
				>
					{item.label}
				</div>
			)}
			value={searchValue}
			onChange={onSearchChange}
			// onSelect={(value) => setSearchValue(value)}
		/>
	);
};

export default Autocomplete;
