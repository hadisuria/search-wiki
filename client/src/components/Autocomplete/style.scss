.input-suggestion {
	display: flex;
	flex-direction: column;
	border: 1px solid grey;
	border-radius: 12px;
	box-sizing: border-box;
}

.search-link {
	margin-left: 20px;
	margin-right: 20px;
	margin-top: 10px;
	margin-bottom: 15px;
	text-align: center;

	&:hover {
		background-color: #eee;
	}
}
