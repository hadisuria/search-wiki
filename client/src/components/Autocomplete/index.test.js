import React from "react";

import { shallow } from "enzyme";
import Autocomplete from "./index";

const render = (props) => shallow(<Autocomplete {...props} />);

describe("Autocomplete component", () => {
	let sut;
	let props;

	describe("without props", () => {
		beforeEach(() => {
			sut = render(props);
		});
		it("should match snapshot", () => {
			expect(sut).toMatchSnapshot();
		});
	});

	// describe("with props", () => {
	// 	beforeEach(() => {
	// 		props = {
	// 			articles: [],
	// 			searchValue: "",
	// 			onSearchChange: jest.fn(),
	// 		};
	// 		sut = render(props);
	// 	});
	// 	it("should match snapshot", () => {
	// 		expect(sut).toMatchSnapshot();
	// 	});
	// });
});
