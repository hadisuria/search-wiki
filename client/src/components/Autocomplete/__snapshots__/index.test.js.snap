// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component without props should match snapshot 1`] = `
<Autocomplete
  autoHighlight={true}
  getItemValue={[Function]}
  inputProps={
    Object {
      "placeholder": "Input text here...",
    }
  }
  isItemSelectable={[Function]}
  menuStyle={
    Object {
      "background": "rgba(255, 255, 255, 0.9)",
      "borderRadius": "3px",
      "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
      "fontSize": "90%",
      "maxHeight": "50%",
      "overflow": "auto",
      "padding": "2px 0",
      "position": "fixed",
    }
  }
  onChange={[Function]}
  onMenuVisibilityChange={[Function]}
  onSelect={[Function]}
  renderInput={[Function]}
  renderItem={[Function]}
  renderMenu={[Function]}
  selectOnBlur={false}
  value=""
  wrapperProps={Object {}}
  wrapperStyle={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "margin": "32px",
      "width": "100%",
    }
  }
/>
`;
